src/taco-cli/cli/kit.ts

```diff
@@ -779,22 +779,20 @@ class Kit extends commands.TacoCommandBase {
 
     /* tslint:disable:member-ordering */
     // tslint doesn't handle this case and considers subcommands as member function
-    public subcommands: commands.ICommand[] = [
+    public subcommands: commands.ISubCommand<Kit>[] = [
         {
             // List kits
             name: "list",
-            run: Kit.list,
-            canHandleArgs(commandData: commands.ICommandData): boolean {
-                return !commandData.remain[0] || commandData.remain[0] && commandData.remain[0].toLowerCase() === "list";
+            run: (command, commandData) => command.list(commandData),
+            canHandleArgs: (command, commandData) => {
```

Comment:
Are you sure about that? When I tested this out the `this` was referring to the class that declared the function, which is `Kit` here.

```
class A {
    public static name: string = "A";
    public name: string = "a";
    public thing = () => {
        console.info(this.name);
    }
    public things = [
        {
            name: "c",
            f: () => console.info(this.name)
        }]
}

var a = new A();
a.thing();
a.things[0].f();
```

prints out `aa` meaning that `this` referred to the instance of the class both times. If the fat arrow function in the `things` list was instead `f(){ console.info(this.name)}` then `this` referred to that object (because the function was invoked from that object).

Now I suspect that the code is currently incorrect with the usage of `this`, and it will refer to `Kit.name` and not the subcommand's name.